### PR TITLE
LBSD-1035 - Registration and management of consumers

### PR DESCRIPTION
**Pull request by m0g**

### CODE REVIEW

File: client/app/scripts/liveblog-syndication/directives/consumer-edit.js [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+                    if (!scope.consumerForm.name.$pristine)
+                        data.name = scope.consumer.name;
+
+                    if (scope.isEditing)
+                        apiQuery = api.save('consumers', scope.origConsumer, data);
+                    else
+                        apiQuery = api.consumers.save(data);
+
+                    apiQuery.then(function(result) {
```
Comment (djbrd-sourcefabric) [inline client/app/scripts/liveblog-syndication/directives/consumer-edit.js]: Handle error

File: client/app/scripts/liveblog-syndication/directives/consumer-edit.js [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+                        notify.pop();
+                        notify.success(gettext('consumer saved.'));
+
+                        if (!scope.isEditing)
```
Comment (djbrd-sourcefabric) [inline client/app/scripts/liveblog-syndication/directives/consumer-edit.js]: Comment what you are doing here

File: client/app/scripts/liveblog-syndication/directives/consumer-list.js [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+                };
+
+                scope.disable = function(e, consumerToRemove) {
+                    e.stopPropagation();
+                    //console.log('disable', consumer);
+
+                    api.consumers.remove(consumerToRemove).then(function(result) {
+                        angular.forEach(scope.consumers, function(consumer, i) {
+                            if (consumer._id == consumerToRemove._id)
+                                scope.consumers.splice(i, 1);
```
Comment (djbrd-sourcefabric) [inline client/app/scripts/liveblog-syndication/directives/consumer-list.js]: Add a break when consumer is found

File: client/app/scripts/liveblog-syndication/views/consumer-list-item.html [hunk 4]
```diff
@@ -0,0 +1,34 @@
+<li ng-repeat="consumer in consumers" ng-click="select(consumer)" ng-class="{selected: consumer === selected, 'inactive' : !consumer.is_enabled || !consumer.is_active}">
```
Comment (djbrd-sourcefabric) [inline client/app/scripts/liveblog-syndication/views/consumer-list-item.html]: Does consumer have is_enabled and is_active functions?

File: client/app/scripts/liveblog-syndication/views/consumer-list-item.html [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+    <figure class="avatar avatar-with-info" ng-class="{'consumer-logged': isLoggedIn(consumer)}">
```
Comment (djbrd-sourcefabric) [inline client/app/scripts/liveblog-syndication/views/consumer-list-item.html]: Consumer will never be logged, I guess, as it is not a user

File: client/app/scripts/liveblog-syndication/views/consumer-list-item.html [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+        <div class="logged-info"></div>
+        <div class="consumer-image">
+            <img sd-consumer-avatar data-consumer="consumer">
+        </div>
+    </figure>
+
+    <i class="admin-label icon-settings" ng-if="consumer.consumer_type == 'administrator'" tooltip="{{ :: 'admin'|translate}}" tooltip-placement="top"></i>
```
Comment (djbrd-sourcefabric) [inline client/app/scripts/liveblog-syndication/views/consumer-list-item.html]: Again, this is legacy from user related code, I think. The consumer cannot be of type administrator. Shouldn't you be checking the logged in user here, for consumer edit rights?

File: client/app/scripts/liveblog-syndication/views/list.html [hunk 7]
```diff
@@ -0,0 +1,79 @@
+<div class="subnav">
+
+    <h3 class="page-nav-title" translate>Users</h3>
```
Comment (djbrd-sourcefabric) [inline client/app/scripts/liveblog-syndication/views/list.html]: Title should be changed

File: client/app/scripts/liveblog-syndication/views/list.html [hunk 8]
```diff
@@ -0,0 +1,79 @@
...
+</div>
+
+<section class="main-section users">
+
+    <div class="preview-layout" ng-class="{closed: !selected.user}">
+        <div class="list-pane">
+            <header>
+                <div sd-searchbar></div>
+                <div class="separator-handler"></div>
+                <div class="online-users">
```
Comment (djbrd-sourcefabric) [inline client/app/scripts/liveblog-syndication/views/list.html]: Appears to be quite a lot of user related code left in this file

File: client/app/scripts/liveblog-syndication/directives/consumer-edit.js (hunk 1 above)
Comment (m0g) [inline client/app/scripts/liveblog-syndication/directives/consumer-edit.js]: Added catch statement

File: client/app/scripts/liveblog-syndication/directives/consumer-list.js (hunk 3 above)
Comment (m0g) [inline client/app/scripts/liveblog-syndication/directives/consumer-list.js]: There is no way to break a forEach statement in js https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/forEach

**Comment by mscam** [comment]
LGTM

File: client/app/scripts/liveblog-syndication/views/consumer-list-item.html [hunk 9]
```diff
@@ -0,0 +1,36 @@
...
+                    <i class="icon-trash true"></i>
+                    <span class="ng-binding">Disable consumer</span>
+                </a>
+            </li>
+        </ul>
+    </div>
+</li>
+
+<li class="alert alert-info" ng-show="consumers.length == null"><p translate>Loading...</p></li>
+<li class="alert alert-info" ng-show="consumers.length == 0"><p translate>Oops! There are no items.</p></li>
```
Comment (nistormihai) [inline client/app/scripts/liveblog-syndication/views/consumer-list-item.html]: Can it be more specific. 
ex: there are no consumers for now.

File: client/spec/consumers_spec.js [hunk 10]
```diff
@@ -0,0 +1,15 @@
+var login = require('../app/scripts/bower_components/superdesk/client/spec/helpers/utils').login,
+    consumersManagement = require('./helpers/pages').consumersManagement;
+
+describe('Consumers', function() {
+    'use strict';
+
+    beforeEach(function(done) {login().then(done);});
+
```
Comment (nistormihai) [inline client/spec/consumers_spec.js]: `editing` and `saving` functionality should be tested as well.

File: client/app/scripts/liveblog-syndication/directives/consumer-list.js (hunk 3 above)
Comment (nistormihai) [inline client/app/scripts/liveblog-syndication/directives/consumer-list.js]: You can throw exceptions or use _.some, but here I don't think should be the case.

File: client/app/scripts/liveblog-syndication/directives/consumer-list.js [hunk 11]
```diff
@@ -0,0 +1,29 @@
...
+                    scope.selected = consumer;
+                };
+
+                scope.disable = function(e, consumerToRemove) {
+                    e.stopPropagation();
+                    //console.log('disable', consumer);
+
+                    api.consumers.remove(consumerToRemove).then(function(result) {
+                        angular.forEach(scope.consumers, function(consumer, i) {
+                            if (consumer._id == consumerToRemove._id)
```
Comment (nistormihai) [inline client/app/scripts/liveblog-syndication/directives/consumer-list.js]: `jshint` with `eqeqeq` will throw error 
please use `grunt hint` command to see if the code standards are ok.

**Pull request merged**